projects/igniteui-angular/src/lib/services/direction/directionality.spec.ts

```diff
@@ -0,0 +1,73 @@
+import { TestBed } from '@angular/core/testing';
+import { Component } from '@angular/core';
+import { IgxDirectionality, DIR_DOCUMENT } from './directionality';
+
+interface FakeDoc {
+    body: { dir?: string };
+    documentElement: { dir?: string };
+}
+
+describe('IgxDirectionality', () => {
+    let fakeDoc: FakeDoc;
+    beforeEach(() => {
+        fakeDoc = {body: {}, documentElement: {}};
+
+        TestBed.configureTestingModule({
+            declarations: [
+                InjectsIgxDirectionalityComponent
+            ],
+            providers: [{provide: DIR_DOCUMENT, useFactory: () => fakeDoc}],
+        }).compileComponents();
+    });
+
+    it('should read dir from html if not specified on the body', () => {
+        const expectedRes = 'rtl';
+        fakeDoc.documentElement.dir = expectedRes;
+
+        const fixture = TestBed.createComponent(InjectsIgxDirectionalityComponent);
```

Comment:
You can put this and the component assign in a `beforeEach` (another, after the async one).

Alternative: You only really need one fixture to test the DI aspect. Might not be a bad idea to compare the result against `TestBed.get(DOCUMENT)` to verify the provider for `DIR_DOCUMENT` does what it should. And then the rest of the tests can easily use `new IgxDirectionality(fakeDoc)`.